docroot/modules/custom/hs_field_helpers/hs_field_helpers.module

```diff
@@ -6,6 +6,74 @@
  */
 
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Layout\LayoutDefinition;
+use Drupal\Core\Render\Element;
+
+/**
+ * Implements hook_preprocess().
+ */
+function hs_field_helpers_preprocess(&$variables, $hook) {
+  // Each layout has its own preprocess hook. So we have to use generic hook.
+  if (isset($variables['layout']) && $variables['layout'] instanceof LayoutDefinition) {
+
+    // This attribute exists when on the layout management page. We don't want
+    // to clean up the regions when on that page because it will break the UI.
+    if (!empty($variables['attributes']['data-layout-update-url'])) {
+      return;
+    }
+
+    foreach (Element::children($variables['content']) as $region) {
+
+      // Some blocks such as a menu block don't have any output, but the layout
+      // still outputs an empty container for the block that has no content.
+      // So lets find out if they are empty, and clean them up.
+      foreach (Element::children($variables['content'][$region]) as $item_id) {
+        if (empty($variables['content'][$region][$item_id]['#theme'])) {
+          unset($variables['content'][$region][$item_id]);
+        }
+      }
+
+      // Clean up empty regions after we've cleaned up the blocks in the region.
+      if (empty(Element::children($variables['content'][$region]))) {
+        unset($variables['content'][$region]);
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function hs_field_helpers_form_field_config_edit_form_alter(&$form, FormStateInterface $form_state, $form_id) {
```

Comment:
This might be a nice patch to submit to d.o

Reply:
I think its related to this issue https://www.drupal.org/project/drupal/issues/953034